js/piwik.js

```diff
@@ -6971,7 +6971,9 @@ if (typeof window.Matomo !== 'object') {
         addEventListener(windowAlias, 'online', function () {
             if (isDefined(navigatorAlias.serviceWorker) && isDefined(navigatorAlias.serviceWorker.ready)) {
                 navigatorAlias.serviceWorker.ready.then(function(swRegistration) {
-                    return swRegistration.sync.register('matomoSync');
+                    if (swRegistration && swRegistration.sync) {
+                        return swRegistration.sync.register('matomoSync');
+                    }
```

Comment:
I'm not very deep into that serviceWorker stuff, but would there be an alternative that could be used if `sync` is not available? Or is it expected that this simply does not work in that case?

Reply:
@sgiehl it is not 100% required for this to work as it's only triggered when status changes from offline to online and our service worker also has an ability to detect this status change eventually it might just take a bit longer.

Reply:
ok. then I guess should be good to merge